docker_deploy/config.go

```diff
@@ -114,3 +117,45 @@ func StringSliceEqual(a, b []string) bool {
 	}
 	return true
 }
+
+// mapsEqual checks if two maps are equal
+func mapsEqual(a, b map[string]interface{}) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for key, valueA := range a {
+		valueB, ok := b[key]
+		if !ok {
+			return false
+		}
+
+		// Different Types
```

Comment:
[opt] Potential simplification using reflect.DeepEquals on interfaces instead of having to go through each type

https://stackoverflow.com/questions/34245932/checking-equality-of-interface